### PR TITLE
GQ-016_ feat: add variables/headers request basic logic

### PR DESCRIPTION
**Pull request by Istiniel**

### CODE REVIEW

File: src/redux/features/AuthSlice/EditorSlice.ts [hunk 1]
```diff
@@ -45,23 +79,43 @@ export const editorSlice = createSlice({
...
+      .addCase(getGqlValueThunk.pending, (state) => {
+        state.status = 'loading'
+      })
+      .addCase(getGqlValueThunk.fulfilled, (state, action) => {
         state.data = action.payload
+        console.log(action.payload)
+        state.status = 'idle'
+      })
+      .addCase(getGqlValueThunk.rejected, (state, action) => {
+        state.status = 'idle'
```
Comment (AnAtoliy-AA) [inline src/redux/features/AuthSlice/EditorSlice.ts:106]: maybe create global const or enum with values, cause you compare with it in component

File: src/redux/features/AuthSlice/EditorSlice.ts [hunk 2]
```diff
@@ -1,38 +1,72 @@
...
 
-export const getGqlValueThunk = createAsyncThunk('editor/getGqlQuery', async (query: string) => {
+export const getGqlValueThunk = createAsyncThunk('editor/getGqlQuery', async (_, thunkAPI) => {
+  const { query, variables, headers } = (thunkAPI.getState() as RootState).editorSlice
+
   const response = await client.query({
     query: gql`
       ${query}
     `,
+
```
Comment (AnAtoliy-AA) [inline src/redux/features/AuthSlice/EditorSlice.ts]: unnecessary empty line

File: src/components/NavBar/index.tsx [hunk 3]
```diff
@@ -35,38 +36,50 @@ const NavBar = () => {
...
+            </li>
+          ) : (
+            <li>
+              <Link
+                href="/auth"
+                className={styles.navBtn}
+                onClick={() => {
+                  signOut(auth)
+                  dispatch(setUser(null))
+                }}
```
Comment (AnAtoliy-AA) [inline src/components/NavBar/index.tsx]: maybe it's better to use handler here?

File: src/components/NavBar/index.tsx [hunk 4]
```diff
@@ -35,38 +36,50 @@ const NavBar = () => {
...
-        )}
+      <div
+        className={classNames(styles.burgerBtn, isOpen ? styles.open : '')}
+        onClick={() => setIsOpen(!isOpen)}
+      >
+        <span></span>
+      </div>
+      <div
+        className={classNames(styles.overlay, isOpen ? styles.openMenu : '')}
+        onClick={() => setIsOpen(!isOpen)}
```
Comment (AnAtoliy-AA) [inline src/components/NavBar/index.tsx]: `const toggleOpen = useCallback(() => { setIsOpen((prev) => !prev }, [])`

`onClick={toggleOpen}`

File: src/components/NavBar/index.tsx [hunk 5]
```diff
@@ -35,38 +36,50 @@ const NavBar = () => {
...
-        {user && (
-          <li>
-            <Link href="/editor" className={styles.navLink}>
-              {capitalizeWord(t('gotomain'))}
-            </Link>
-          </li>
-        )}
+      <div
+        className={classNames(styles.burgerBtn, isOpen ? styles.open : '')}
+        onClick={() => setIsOpen(!isOpen)}
```
Comment (AnAtoliy-AA) [inline src/components/NavBar/index.tsx]: `const toggleOpen = useCallback(() => { setIsOpen((prev) => !prev }, [])`

`onClick={toggleOpen}`

File: src/redux/features/AuthSlice/EditorSlice.ts (hunk 1 above)
Comment (Istiniel) [reply src/redux/features/AuthSlice/EditorSlice.ts:106]: will be fixed/refactor in following commits

File: src/redux/features/AuthSlice/EditorSlice.ts (hunk 2 above)
Comment (Istiniel) [reply src/redux/features/AuthSlice/EditorSlice.ts]: will be fixed/refactor in following commits

File: src/components/NavBar/index.tsx (hunk 5 above)
Comment (Istiniel) [reply src/components/NavBar/index.tsx]: will be fixed/refactor in following commits

File: src/components/NavBar/index.tsx (hunk 4 above)
Comment (Istiniel) [reply src/components/NavBar/index.tsx]: will be fixed/refactor in following commits

File: src/UI/EditorButtons/index.tsx [hunk 6]
```diff
@@ -32,14 +29,18 @@ export const SwitchButton: React.FC<ISwitchButton & React.HTMLProps<HTMLButtonEl
   )
 }
 
+interface IStartButton {
+  onClickHandler?: () => void
```
Comment (AnAtoliy-AA) [inline src/UI/EditorButtons/index.tsx]: you can use `voidFunction`

**Review by AnAtoliy-AA — APPROVED** [review]

**Pull request merged**